template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/impala/ImpalaVolumeConfigProvider.java

```diff
@@ -38,20 +40,49 @@ public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, HostgroupV
                         .mapToInt(volume -> volume.getVolumeSize())
                         .min()
                         .orElse(0);
+                int volumeCount = 0;
+                int temporaryStorageVolumeCount = 0;
+                int temporaryStorageVolumeSize = 0;
+                if (hostGroupView != null) {
+                    volumeCount = hostGroupView.getVolumeCount();
+                    if (hostGroupView.getTemporaryStorageVolumeCount() != null) {
+                        temporaryStorageVolumeCount = hostGroupView.getTemporaryStorageVolumeCount();
+                        temporaryStorageVolumeSize = hostGroupView.getTemporaryStorageVolumeSize();
+                    }
+                }
 
                 List<ApiClusterTemplateConfig> configs = new ArrayList<>();
-                configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
-                        buildVolumePathStringZeroVolumeHandled(hostGroupView.getVolumeCount(), "impala/scratch")));
+
+                if (source.getCloudPlatform().equalsIgnoreCase("AWS") && checkTemporaryStorage(hostGroupView, temporaryStorageVolumeCount)) {
+                    configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
+                            buildEphemeralVolumePathString(temporaryStorageVolumeCount, "impala/scratch")));
+                } else {
+                    configs.add(config(IMPALA_SCRATCH_DIRS_PARAM,
+                            buildVolumePathStringZeroVolumeHandled(volumeCount, "impala/scratch")));
+                }
 
                 if (minAttachedVolumeSize > MIN_ATTACHED_VOLUME_SIZE_TO_ENABLE_CACHE_IN_GB
-                        && hostGroupView.getVolumeCount() > 0) {
+                        && volumeCount > 0) {
                     long dataCacheCapacityInBytes = VolumeUtils
-                            .convertGBToBytes(Math.min(minAttachedVolumeSize / 2, MAX_IMPALA_DATA_CACHE_SIZE_IN_GB / hostGroupView.getVolumeCount()));
+                            .convertGBToBytes(Math.min(minAttachedVolumeSize / 2, MAX_IMPALA_DATA_CACHE_SIZE_IN_GB / volumeCount));
```

Comment:
It's unnecessary to declare and calculate dataCacheCapacityInBytes here as it's not used outside of the else branch in the if-else beneath (line 80). Please move it there.

Reply:
Okay, I will move it.